Cargo.toml

```diff
@@ -1,3 +1,8 @@
+[workspace]
+members = [
+    "poseidon-merkle"
+]
```

Comment:
Following the naming below, shouldn't this crate be called `dusk-merkle-poseidon` or `dusk-poseidon-merkle`?

Reply:
I'm somewhat conflicted on the "who needs to know what". Should a lower level primitive (imo) like a hashing function be aware of a generic merkle tree implementation that can take any type as a node? No.

So Poseidon does belong in here, versus the other way around.

Should Poseidon be explicitly in Merkle? Seems icky to me. The approach you suggest here is one I also suggested to @moCello a while back. While it seems somewhat convoluted, it does provide the right abstraction and boxing of responsibilities. 

`dusk-merkle-poseidon` seems the most logical option to me, as we can use a scheme like this: `dusk-merkle-{hash/mapping func}`.

Reply:
I'm not particularly invested in the names, but `poseidon_merkle` is still free on crates.io and I thought that, if possible, we always want to have names without the `dusk` prefix. Also, now that they are separate crates, they don't necessarily need to follow the same naming convention.

Reply:
@ureeves in your suggestion above:
```
Cargo.toml
lib/
  poseidon/
      src/
      Cargo.toml
  merkle/
      src/
      Cargo.toml
```
1. Why do you think that having the two libraries on the same level is better?
2. Why are they nested in a 'lib/' directory?

Reply:
Any comments on what i wrote in the [issue description](https://github.com/dusk-network/merkle/issues/58#issuecomment-1602245706) regarding the opening gadget?

Reply:
> Any comments on what i wrote in the [issue description](https://github.com/dusk-network/merkle/issues/58#issuecomment-1602245706) regarding the opening gadget?

Since we're going to have to deal with separating the crates anyway, we'll have to deal with this limitation. Unless we want to introduce a trait so that we can attach such a functionality to the `Opening` struct directly, but I think that's a bad idea. For now.